kotlinx-coroutines-core/jvm/src/CoroutineContext.kt

```diff
@@ -48,6 +49,78 @@ internal actual inline fun <T> withCoroutineContext(context: CoroutineContext, c
     }
 }
 
+/**
+ * Executes a block using a context of a given continuation.
+ */
+internal actual inline fun <T> withContinuationContext(continuation: Continuation<*>, countOrElement: Any?, block: () -> T): T {
+    val context = continuation.context
+    val oldValue = updateThreadContext(context, countOrElement)
+    val undispatchedCompletion = if (oldValue !== NO_THREAD_ELEMENTS) {
```

Comment:
`Nothing` is inferred in type parameter, but it doesn't affect correctness.
Filed https://youtrack.jetbrains.com/issue/KT-43707